File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode temp = head;
        ListNode newNode = null;
         ListNode root = null;
        while (temp != null) {
            while (temp.next != null && temp.next.val == temp.val) {
                temp=temp.next;
            }
            ListNode node = new ListNode();
                node.val = temp.val;
            if(newNode == null) {
                newNode = node;
                root = node;
            }else {
                newNode.next = node;
                newNode = newNode.next;
            }
            temp = temp.next;
        }
        return root;
    }
}

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head || !head->next)
            return head;
        ListNode* temp = head;
        while(temp)
        {
            while(temp->next && temp->val == temp->next->val)
            {
                ListNode* next = temp->next;
                temp->next = temp->next->next;
                delete next;
            }
            temp = temp->next;
        }
        return head;
    }
};